napari/layers/shapes/_tests/test_shapes_mouse_bindings.py

```diff
@@ -17,20 +18,27 @@
 )
 
 
-@pytest.fixture
-def Event():
-    """Create a subclass for simulating vispy mouse events.
+@dataclass
+class Event:
+    """Create a subclass for simulating vispy mouse events."""
 
-    Returns
-    -------
-    Event : Type
-        A new tuple subclass named Event that can be used to create a
-        NamedTuple object with fields "type", "is_dragging", and "modifiers".
-    """
-    return collections.namedtuple(
-        'Event',
-        field_names=['type', 'is_dragging', 'modifiers', 'position', 'pos'],
-    )
+    type: str
+    is_dragging: bool = False
+    modifiers: List[str] = field(default_factory=list)
+    position: Union[Tuple[int, int], Tuple[int, int, int]] = (
+        0,
+        0,
+    )  # world coords
+    pos: np.ndarray = field(
+        default_factory=lambda: np.zeros(2)
+    )  # canvas coords
+    view_direction: Optional[List[float]] = None
+    up_direction: Optional[List[float]] = None
+    dims_displayed: List[int] = field(default_factory=lambda: [0, 1])
+
+
+def read_only_event(*args, **kwargs):
```

Comment:
```suggestion
def read_only_mouse_event(*args, **kwargs):
```
same as above 

Maybe event it may in future be extracted to fixture to simplify reuse in future versions.